oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,294 @@
+##################
+OEP-44: Changelogs
+##################
+
+.. list-table::
+
+   * - OEP
+     - :doc:`OEP-0044 </oeps/oep-0044>`
+   * - Title
+     - Changelogs
+   * - Last Modified
+     - 2020-03-19
+   * - Authors
+     - Robert Raposa
+   * - Arbiter
+     - ???
+   * - Status
+     - Draft
+   * - Type
+     - Best Practice
+   * - Created
+     - 2020-03-19
+   * - Review Period
+     - ???
+
+.. contents::
+   :local:
+   :depth: 3
+
+Context
+=======
+
+There are a variety of positions taken in Open edX by individual developers and for each repository regarding how and if and when and why to include changelogs, and how to manage software versions. This OEP will provide best practices for managing both changelogs and versions.
+
+Background
+==========
+
+A Changelog Definition
+----------------------
+
+A very useful reference is `keepachangelog.com`_, although this OEP does depart from this in some significant ways.  The following questions and answers are copied directly from `keepachangelog.com`_
+
+What is a changelog?
+~~~~~~~~~~~~~~~~~~~~
+
+A changelog is a file which contains a curated, chronologically ordered list of notable changes for each version of a project.
+
+Why keep a changelog?
+~~~~~~~~~~~~~~~~~~~~~
+
+To make it easier for users and contributors to see precisely what notable changes have been made between each release (or version) of the project.
+
+Who needs a changelog?
+~~~~~~~~~~~~~~~~~~~~~~
+
+People do. Whether consumers or developers, the end users of software are human beings who care about what's in the software. When the software changes, people want to know why and how.
```

Comment:
Is there some way to measure success? It would be easy to count the number of CHANGELOG.rst files, but are the changelogs good? Are they accomplishing the goal of adequately educating the readers? What can we do to assess our progress toward good changelogs?

Reply:
Could this OEP have a rubric for the kinds of information that should be included in a changelog entry?

Reply:
1. I'm sure there are ways to measure success, but I don't see how to do it without either automation or surveying the audiences of changelogs, and this hasn't been a priority. My first priority was to provide clarity so someone could simply know if they were doing or not doing what was expected.  I imagine we might be able to add to the new Repository Health Dashboard some metric related to the number of entries vs the number of merges, etc., but I don't want to block this PR on this.  I could add an "Open Questions" or "Next Steps" section that acknowledges this direction we'd like to go in.
2. I'd love help adding a rubric if you have ideas.  I think what I added gives a strong indication of the types of changes to consider, but happy to provide something else or in addition.

Reply:
I think a good measure for success would be for Open edX newcomers to be able to access, read, understand and link to the changelog. I understand this is not an objective measure, but after all we are building something for humans here.

Reply:
For measuring success, we could self-audit.  Three months after ratifying the OEP, take a survey of repos.  How many claim to conform?  Of those, read the changelogs: do we think they have the right information?

Reply:
I can't mark this as resolved but yes, I'm very satisfied.